app/models/manageiq/providers/azure_stack/inventory/parser/network_manager.rb

```diff
@@ -3,11 +3,59 @@ def parse
     log_header = "Collecting data for EMS : [#{collector.manager.name}] id: [#{collector.manager.id}]"
     $azure_stack_log.info("#{log_header}...")
 
+    cloud_networks
+    network_ports
     security_groups
 
     $azure_stack_log.info("#{log_header}...Complete")
   end
 
+  def cloud_networks
+    collector.networks.each do |network|
+      persister.cloud_networks.build(
+        :name    => network.name,
+        :ems_ref => network.id.downcase
+      )
+      if network.subnets
+        cloud_subnets(network)
+      end
+    end
+  end
+
+  def cloud_subnets(network)
+    network.subnets.each do |subnet|
+      persister.cloud_subnets.build(
+        :name            => subnet.name,
+        :ems_ref         => subnet.id.downcase,
+        :cloud_network   => persister.cloud_networks.lazy_find(network.id.downcase),
+        :security_groups => build_security_groups(subnet)
+      )
+    end
+  end
+
+  def network_ports
+    collector.network_ports.each do |port|
+      persister.network_ports.build(
+        :name            => port.name,
+        :ems_ref         => port.id.downcase,
+        :mac_address     => port.mac_address,
+        :device          => port.virtual_machine ? persister.vms.lazy_find(port.virtual_machine.id.downcase) : nil,
+        :security_groups => build_security_groups(port)
```

Comment:
I'm not sure we need utility function here because we're just filling in a simple relation. I'd go with a simple IF statement at the top instead:

```ruby
security_groups = []
if (security_group_id = port.network_security_group&.id&.downcase)
  security_groups = [persister.security_groups.lazy_find(security_group_id)]
end
```